### ISSUE TITLE
ANTIQUE Dataset Tokenisation

### ISSUE THREAD

**Issue by divyakyatam** [tensorflow/ranking#206]
Hi TFR Team,

I tried creating tf-records from raw ANTIQUE dataset, but I couldn't reproduce similar results. Accuracies are pretty low.

Could you please share with us on what kind of tokenisation you have used to create document and query tokens

Thanks.

**Comment by ramakumar1729** [comment]
We used BERT's [wordpiece](https://github.com/google-research/bert#tokenization) tokenizer.

**Comment by divyakyatam** [comment]
Thank You @ramakumar1729.

**Comment by divyakyatam** [comment]
@ramakumar1729  May I also please how you have created vocab.txt, Before or after tokenising. As in the definition of full tokeniser vocal file is passed... `tokenizer = tokenization.FullTokenizer(
    vocab_file=vocab_file, do_lower_case=True)
`

Could you also please help me with this.

Thanks in advance

**Comment by divyakyatam** [comment]
@ramakumar1729  Thanks again

**Comment by divyakyatam** [comment]
Hey @ramakumar1729, I have implemented both the tokenisers mentioned below and still couldn't reproduce the results. `from bert import bert_tokenization`  `tokenizer = bert_tokenization.FullTokenizer(
    vocab_file=vocab_file, do_lower_case=True)`.    and  `from transformers import BertTokenizer`   `tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')` 

Have you got any suggestions here to improve the accuracy?

Thanks

**Comment by ramakumar1729** [comment]
@divyakyatam : Can you compare the data you generated with data we provide at [http://ciir.cs.umass.edu/downloads/Antique/tf-ranking/](http://ciir.cs.umass.edu/downloads/Antique/tf-ranking/).

**Comment by divyakyatam** [comment]
Hi @ramakumar1729,

I did compare with the files provided [http://ciir.cs.umass.edu/downloads/Antique/tf-ranking/.](here)

**Comment by divyakyatam** [comment]
Hi @ramakumar1729 ,

I have taken a look at this [issue](https://github.com/tensorflow/ranking/issues/192) and understand that we should generate vocab.txt after applying tokenisation. But, above you have mentioned you have used `vocal.txt` from [bert github repo](https://github.com/google-research/bert#bert)

Thanks

**Comment by divyakyatam** [comment]
Hi,

I also found difference in results when running notebook version of TF-Ranking are higher compared to script version

`handling_sparse_features.ipynb` -notebook version

`({'global_step': 15000,
  'labels_mean': 1.9630322,
  'logits_mean': 1.8256326,
  'loss': -0.8472472,
  'metric/ndcg@1': 0.705,
  'metric/ndcg@10': 0.8444919,
  'metric/ndcg@3': 0.7445483,
  'metric/ndcg@5': 0.78524727},
 [])`

`tf_ranking_tfrecord.py`  - script version

`Saving dict for global step 15000: global_step = 15000, 
 labels_mean = 1.8611765, 
 logits_mean = 1.9152381, 
 loss = -0.79547465, 
 metric/arp = 8.963338, 
 metric/ndcg@1 = 0.68303573, 
 metric/ndcg@10 = 0.80916977, 
 metric/ndcg@3 = 0.69290155, 
 metric/ndcg@5 = 0.7276103, 
 metric/ordered_pair_accuracy = 0.6492308, 
 metric/weighted_ndcg@1 = 0.68303573, 
 metric/weighted_ndcg@10 = 0.80916977, 
 metric/weighted_ndcg@3 = 0.69290155, 
 metric/weighted_ndcg@5 = 0.7276103`

Could you please explain why is that difference ?

Thanks

**Comment by divyakyatam** [comment]
Hi @ramakumar1729,

While comparing tf_records, what I could understand was `**How did you decide you decide on number of documents for each query** `

If you could please answer this, it would be really very helpful.

Thanks

**Comment by ramakumar1729** [comment]
@divyakyatam : For a custom dataset, it is better to create your own vocabulary. Note that the purpose of this demo is to show reasonable training curves on Antique dataset, not the best possible performance of a neural network model. 

One difference I see between the script and demo version is that the `list_size` is 50 in the demo and set to `None` in the script. `None` indicates variable number of documents can be present per query and are processed, whereas a fixed list size truncates or pads the list. This can be causing the difference. Note that for fair evaluation, the `eval_input_fn` should have the same `list_size`.

Afaik, Antique dataset has a variable number of documents per query. We do not do any sampling or truncation as such. Let me know if you have any other questions.

**Comment by divyakyatam** [comment]
Thank you @ramakumar1729 for the support.
